### PR TITLE
[KEYCLOAK-7552] - Improvements to AuthZ Quickstarts and Docs

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

File: authorization_services/topics/policy-js-policy.adoc [hunk 1]
```diff
@@ -64,6 +83,66 @@ if (identity.hasClientRole('my-client', 'my-client-role')) {
 }
 ```
 
+=== Checking for roles granted to an user
+You can also check for realm roles granted to an user:
```
Comment (matthewhelmke) [inline authorization_services/topics/policy-js-policy.adoc]: s/You can also/To

File: authorization_services/topics/policy-js-policy.adoc [hunk 2]
```diff
@@ -64,6 +83,66 @@ if (identity.hasClientRole('my-client', 'my-client-role')) {
...
+var realm = $evaluation.getRealm();
+
+if (realm.isUserInClientRole('marta', 'my-client', 'some-client-role')) {
+    $evaluation.grant();
+}
+```
+
+=== Checking for roles granted to a group
+You can also check for realm roles granted to an user:
+
```
Comment (matthewhelmke) [inline authorization_services/topics/policy-js-policy.adoc]: s/You can also/To

File: authorization_services/topics/policy-js-policy.adoc [hunk 3]
```diff
@@ -64,6 +83,66 @@ if (identity.hasClientRole('my-client', 'my-client-role')) {
...
+```javascript
+var realm = $evaluation.getRealm();
+
+if (realm.isGroupInRole('/Group A/Group D', 'role-a')) {
+    $evaluation.grant();
+}
+```
+
+=== Pushing arbitrary claims to the resource server
+You can push arbitrary claims to the resource server in order to provide additional information on how permissions should be
```
Comment (matthewhelmke) [inline authorization_services/topics/policy-js-policy.adoc]: s/You can/To

**Review by matthewhelmke — CHANGES_REQUESTED** [review]
Looks good, just small changes being requested.

**Comment by pedroigor** [comment]
It seems that errors in travis are unrelated

File: authorization_services/topics/enforcer-keycloak-enforcement-filter.adoc [hunk 4]
```diff
@@ -61,7 +61,8 @@ for all resources associated with the resource server being protected. In this c
 +
 ** *user-managed-access*
 +
-Specifies that the adapter uses the UMA protocol. If specified, the adapter queries the server for permission tickets and return them to clients according to the UMA specification. If not specified, the adapter relies on the requesting party token (RPT) sent to the server to enforce permissions.
+Specifies that the adapter uses the UMA protocol. If specified, the adapter queries the server for permission tickets and return them to clients according to the UMA specification. If not specified, the policy enforcer will be able to enforce permissions based on regular access tokens or RPTs. In this case,
```
Comment (matthewhelmke) [inline authorization_services/topics/enforcer-keycloak-enforcement-filter.adoc]: "...adapter queries the server for permission tickets and returns them..."

File: authorization_services/topics/service-protection-policy-api.adoc [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+	"name": "Any people manager",
+	"description": "Allow access to any people manager",
+	"scopes": ["read"],
+	"condition": "if (isPeopleManager()) {$evaluation.grant()}"
+}'
+```
+
+It is also possible to set any combination of these access control mechanisms.
+
+To update an existing permission, you can send a HTTP PUT request as follows:
```
Comment (matthewhelmke) [inline authorization_services/topics/service-protection-policy-api.adoc]: To update an existing permission, send an HTTP PUT request as follows:

File: authorization_services/topics/service-protection-policy-api.adoc [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    "decisionStrategy": "UNANIMOUS",
+    "owner": "7e22131a-aa57-4f5f-b1db-6e82babcd322",
+    "roles": [
+        "user"
+    ]
+}'
+```
+== Removing a Permission
+
+To remove a permission associated with a resource, you can send an HTTP DELETE request as follows:
```
Comment (matthewhelmke) [inline authorization_services/topics/service-protection-policy-api.adoc]: remove "you can"

File: authorization_services/topics/service-protection-policy-api.adoc [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+```bash
+curl -X DELETE \
+  http://localhost:8180/auth/realms/photoz/authz/protection/uma-policy/{permission_id} \
+  -H 'Authorization: Bearer '$access_token
+```
+
+== Querying Permission
+
+To query the permissions associated with a resource, you can send an HTTP GET request as follows:
```
Comment (matthewhelmke) [inline authorization_services/topics/service-protection-policy-api.adoc]: remove "you can"

File: authorization_services/topics/service-protection-policy-api.adoc [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+```bash
+http://${host}:${post}/auth/realms/${realm}/authz/protection/uma-policy?resource={resource_id}
+```
+
+To query the permissions given its name, you can send an HTTP GET request as follows:
```
Comment (matthewhelmke) [inline authorization_services/topics/service-protection-policy-api.adoc]: remove "you can"

File: authorization_services/topics/service-protection-policy-api.adoc [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+```bash
+http://${host}:${post}/auth/realms/${realm}/authz/protection/uma-policy?name=Any people manager
+```
+
+To query the permissions associated with a specific scope, you can send an HTTP GET request as follows:
```
Comment (matthewhelmke) [inline authorization_services/topics/service-protection-policy-api.adoc]: remove "you can"

File: authorization_services/topics/service-protection-policy-api.adoc [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+```bash
+http://${host}:${post}/auth/realms/${realm}/authz/protection/uma-policy?scope=read
+```
+
+To query all permissions, you can send an HTTP GET request as follows:
```
Comment (matthewhelmke) [inline authorization_services/topics/service-protection-policy-api.adoc]: remove "you can"

File: authorization_services/topics/service-protection-policy-api.adoc [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+```bash
+http://${host}:${post}/auth/realms/${realm}/authz/protection/uma-policy
+```
+
+When querying the server for permissions you can also use parameters `first` and `max` results to limit the result.
```
Comment (matthewhelmke) [inline authorization_services/topics/service-protection-policy-api.adoc]: remove "you can also"

**Review by matthewhelmke — CHANGES_REQUESTED** [review]

**Comment by matthewhelmke** [comment]
@pedroigor did all my requested changes make it in?

**Comment by pedroigor** [comment]
@matthewhelmke, yes :)

**Review by matthewhelmke — DISMISSED** [review]

File: authorization_services/topics/service-protection-resources-api-papi.adoc [hunk 12]
```diff
@@ -16,3 +16,127 @@ This endpoint provides operations outlined as follows (entire path omitted for c
 * List resource set descriptions: GET /resource_set
 
 For more information about the contract for each of these operations, see https://docs.kantarainitiative.org/uma/wg/oauth-uma-federated-authz-2.0-09.html#reg-api[UMA Resource Registration API].
+
+== Creating a Resource
+
+To create a resource you must send a HTTP POST request as follows:
+
```
Comment (matthewhelmke) [inline authorization_services/topics/service-protection-resources-api-papi.adoc]: ...an HTTP...

File: authorization_services/topics/service-protection-resources-api-papi.adoc [hunk 13]
```diff
@@ -16,3 +16,127 @@ This endpoint provides operations outlined as follows (entire path omitted for c
...
+         "read-public",
+         "post-updates",
+         "read-private",
+         "http://www.example.com/scopes/all"
+      ]
+  }'
+```
+
+By default, the owner of a resource is the resource server. If you want to define a different owner such as an
+specific user, you can send a request as follows:
```
Comment (matthewhelmke) [inline authorization_services/topics/service-protection-resources-api-papi.adoc]: ...different owner, such as...

**Review by matthewhelmke — CHANGES_REQUESTED** [review]
Looks good, thanks! Two very small changes being requested.

**Comment by pedroigor** [comment]
@matthewhelmke, I have added the last two commits after your last review. They are adding more references to other parts of the documentation.

**Review by matthewhelmke — APPROVED** [review]

**Comment by matthewhelmke** [comment]
Thanks, @pedroigor

**Pull request merged**